functorch/csrc/DynamicLayer.cpp

```diff
@@ -254,6 +254,7 @@ void dynamicLayerFrontFallback(const c10::OperatorHandle& op, torch::jit::Stack*
   //   exclude = exclude.remove(DispatchKey::Batched);
   } else if (layer.key() == kBatchedKey) {
     exclude = exclude.remove(kBatchedKey);
+    c10::impl::tls_set_dispatch_key_included(DispatchKey::FuncTorchVmapMode, true);
```

Comment:
After exiting the dynamic layer system, is FuncTorchVmapMode set? I'm a little worried because I don't see who actually turns FuncTorchVmapMode off. (Although [we do reset the local dispatch keyset in DynamicLayerBack](https://github.com/zou3519/functorch/blob/5d7553b667d55329b38993068f04128ecc8eee33/functorch/csrc/DynamicLayer.cpp#L349-L350), the SaveLocalDispatchKeySet actually restores the state when we're done with the current chain of dispatch.

So for example:
```
vmap(torch.sin)(x)
# Is FuncTorchVmapMode still on here? It shouldn't be but I am worried it is in the current design
torch.randn(3, 3)
```

If this is the case, an easy solution is to have _vmapmode_decrement_nesting always attempt to remove FuncTorchVmapMode from the local include set